lib/sales_analyst.rb

```diff
@@ -80,11 +86,129 @@ def standard_deviation(values, mean)
     std_dev.round(2)
   end
 
+  def average_invoices_per_merchant
+    average = all_invoices.length.to_f / all_merchants.length
+    average.round(2)
+  end
+
+  def num_of_invoices_per_merchant
+    all_merchants.each_with_object({}) do |merchant, total_per_merchant|
+      total_invoices = all_invoices.count do |invoice|
+        invoice.merchant_id == merchant.id
+      end
```

Comment:
⛔️ We've got a find_all_by_merchant_id method for invoices (invoice repo) already that you can use here @NoahZinter -- then you can just .length on that

Reply:
Good catch! Thanks

Reply:
Looks like we'll have to wait on that refactor until we have more access to repo methods from our inheritance refactor